test/pupil/features/step_definitions/device_steps.rb

```diff
@@ -0,0 +1,17 @@
+Then(/^I should see device information populated in local storage$/) do
+  device_info = JSON.parse(page.evaluate_script('window.localStorage.getItem("device");'))
+  expect(device_info['battery'].keys).to eql ["isCharging", "levelPercent", "chargingTime", "dischargingTime"]
```

Comment:
Battery stuff is Chrome-specific, fyi

Reply:
We are currently only running in headless chrome, so should be consistent, when we begin running on different browsers it may become a issue.

Reply:
@GuyHarwood yep can do ill add it now